cmd/cmd_test.go

```diff
@@ -0,0 +1,51 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+const cfgHomeKey = "EXERCISM_CONFIG_HOME"
+
+type CommandTest struct {
+	App            *cobra.Command
+	Cmd            *cobra.Command
+	InitFn         func()
+	TmpDir         string
+	Args           []string
+	OriginalValues struct {
+		ConfigHome string
+		Args       []string
+	}
+}
+
```

Comment:
I didn't create a `NewCommandTest` method, because the arguments would end up being long and messy, and I prefer to have them all on separate lines.

To initialize one of these it needs 3 of the fields: `Cmd` (e.g. `downloadCmd`), `InitFn` (e.g. `initDownloadCmd`), and `Args` (the faked out os.Args).